src/utils/tokenizers.py

```diff
@@ -53,8 +53,16 @@ def detokenize(self, tokens):
         '''
         return [self._detokenizer.unescape_xml(t) for t in tokens]
 
+class BertTokenizer(Tokenizer):
+    def __init__(self, bert_model_name):
+        from pytorch_pretrained_bert import BertTokenizer as HFBertTokenizer
+        do_lower_case = bert_model_name.endswith('uncased')
+        self._tokenizer = HFBertTokenizer.from_pretrained(bert_model_name, do_lower_case=do_lower_case)
+
+    def tokenize(self, sentence):
+        return self._tokenizer.tokenize(sentence)
 
 AVAILABLE_TOKENIZERS = {
     "OpenAI.BPE": OpenAIBPETokenizer(),
-    "MosesTokenizer": MosesTokenizer()
+    "MosesTokenizer": MosesTokenizer(),
```

Comment:
I think it's cleaner and clearer to have all the BERT tokenizers listed here, even if you are handling them specially later

Reply:
To avoid initializing BERT tokenizer on each process sentence call, we could just construct all four(?) BERT tokenizers here, but that might be annoying if you're not using BERT

Reply:
See note above on using a cache.